### ISSUE TITLE
MAG-BENCH-002: freeze cross-family Fe-Co-X benchmark and contamination boundary

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#5115]
## Mission

Define one reusable retrospective benchmark contract for rare-earth-free `Fe-Co-X` families before additional literature/data are consumed by candidate generation or model tuning.

This is the contamination firewall between:

- historical knowledge used to build/train/search;
- retrospective targets used only for evaluation;
- future prospective candidates/results that were unavailable at the frozen cutoff.

## Core theorem

```text
publication is recent
!= target was unseen by the system

formula absent from one corpus
!= structure/property target uncontaminated

random train/test split
!= chemistry-family generalization
!= structure-family generalization
!= temporal generalization
!= prospective discovery
```

## Initial family registry

Start with separately identified families rather than one pooled magnet table:

- Fe-Co-Zr
- Fe-Co-S
- Fe-Co-P
- Fe-Co-Si
- Fe-Co-B
- Fe-Co-C

Later quaternaries such as Fe-Co-Zr-Mn require a separate profile/version rather than silently widening the benchmark universe.

A family identity binds its allowed element set and composition-space semantics. Do not infer family solely from formula text after scoring.

## `MagnetBenchmarkKnowledgeCutoffV1`

Freeze before target ingestion:

- benchmark/profile revision;
- exact UTC cutoff instant/date policy;
- allowed source classes;
- provider/database snapshot identities available before cutoff;
- literature/source snapshot identities available to candidate generation before cutoff;
- exact model/training dataset versions and declared training cutoffs where knowable;
- source artifacts whose training inclusion is unknown;
- exact search-code/model/config identities covered by the cutoff;
- contamination-policy revision;
- deterministic cutoff identity.

Unknown training exposure must remain `Unknown`, not be assumed clean.

## Target visibility states

Use explicit target-level states such as:

- `KnownBeforeCutoff`
- `StructureKnownPropertyHidden`
- `CompositionKnownStructureHidden`
- `PostCutoffPublication`
- `TrainingExposureUnknown`
- `PotentialSemanticLeakage`
- `EligibleRetrospectiveHoldout`
- `EligibleProspectiveOnly`

Do not collapse visibility to one `is_contaminated` boolean.

## Scientific target identity

For benchmark targets, bind the exact scientific question separately from expected answers.

At minimum:

- exact MAT-007 subject/structure identity where authority permits;
- family identity;
- property ID;
- MAT-008 condition signature;
- method/evidence class expected for comparison;
- target-source authority/disclosure identity;
- expected-answer artifact separately;
- publication/source date and capture date;
- visibility/contamination classification.

Expected numeric answers must not participate in candidate/search identity.

## Split axes

Every benchmark result should declare exactly which split theorem it evaluates. Support at least:

### `CompositionFamilyHoldout`

Hold out an entire X family where possible.

### `StructureFamilyHoldout`

Keep crystallographically related structures/prototypes together. Exact grouping rule must be frozen from structure identity/prototype/symmetry evidence rather than chosen after observing performance.

### `TemporalHoldout`

Targets published after a frozen knowledge cutoff, subject to training-exposure audit.

### `CompoundHoldoutWithinFamily`

Useful for interpolation benchmarking but must not be advertised as family-generalization evidence.

### `PropertyHoldout`

Structure/composition may be known while the exact quantitative property label is withheld. Keep this distinct from new-material discovery.

A row may belong to multiple descriptive slices, but one result must name its primary preregistered split profile.

## Leakage graph

Contamination should be graph-shaped, not simple formula matching. Track possible relations such as:

```text
same exact structure
same normalized composition
same prototype/structure family
same database material/entry
same source publication
same derived database lineage
same training dataset
same model family
same property label
same candidate-generator feature source
```

If a target structure appears in training but `K1` is withheld, that is a property-holdout theorem, not unseen-structure discovery.

## Negative / failure targets

Benchmark sets must retain scientifically important negatives, not only published winners:

- unstable / clearly above-hull candidates;
- dynamically unstable structures where qualified evidence exists;
- low-anisotropy/high-moment tradeoffs;
- high-anisotropy/low-moment tradeoffs;
- magnetic-state ambiguity/nonconvergence;
- reported candidates that later calculations contradict;
- failed/nonconverged local reproductions.

A benchmark made only of headline successful materials cannot measure useful precision/selectivity.

## Property vector

Keep dimensions separate:

- formation energy;
- exact-reference-set `ΔE_hull`;
- harmonic dynamic-stability state;
- magnetic-state confidence;
- `Ms` / `Js`;
- directional MAE and/or derived `K1`;
- `Tc` evidence where method authority permits;
- density;
- supply criticality / elemental burden;
- raw-material/process-cost evidence;
- later microstructure-bound `Hc`, `Br`, `(BH)max` only through MAG-002's extrinsic bridge.

No hidden weighted `magnet score`.

### Retrospective reproduction

The system is allowed to use only the frozen pre-cutoff knowledge view. Evaluate whether known post-cutoff/held-out scientific results are reproduced under the declared split.

### Search enrichment

Freeze a candidate universe and acquisition/ranking rule, then evaluate enrichment of desirable high-fidelity outcomes relative to controls.

### Prospective deferred evaluation

Freeze candidate commitments **before** high-fidelity answers or later publications are observed. This is a distinct lineage from retrospective benchmarking.

Never reinterpret a retrospective target as prospective because the implementation did not explicitly look up the answer during one run.

## Control allocations for search campaigns

Before expensive MAT-CALC/DFT evaluation freeze:

- model top-K;
- random controls;
- structure/chemistry diversity controls;
- uncertainty/OOD controls;
- optional conventional descriptor/baseline controls;
- compute budget;
- stopping rule.

Report outcome enrichment per high-fidelity compute, not only best discovered value.

## Cross-family publication ingestion rule

For Fe-Co-S/P/Si/B/C benchmark papers:

1. capture exact source artifact/authority first;
2. classify publication and target visibility against the frozen cutoff;
3. freeze target identities/expected-answer custody;
4. only then allow evaluation code to read expected answers;
5. candidate generation/search must not ingest holdout answer artifacts.

If a publication is used to improve the search model, it becomes training/background evidence and cannot remain a clean target under the same benchmark lineage.

## Metamorphic/adversarial tests

1. same formula + different structure -> distinct target identity;
2. same structure + withheld property -> property holdout, not unseen structure;
3. target published post-cutoff but present in pretraining dataset -> contamination/unknown, not clean temporal holdout;
4. changing cutoff changes benchmark identity;
5. moving one family from train to holdout changes split identity;
6. random row splitting of related temperature/property records cannot satisfy family/structure holdout;
7. target answer artifact inserted into candidate feature source -> benchmark invalidation;
8. aliased database IDs resolving to same scientific target cannot appear on both sides of the split;
9. failed/nonconverged target cannot silently disappear from denominator;
10. source exposure classified `Unknown` cannot be promoted to `Clean` without new evidence;
11. model retraining after target reveal creates a new benchmark/model lineage;
12. expected numeric answer mutation changes answer artifact but not target-question identity.

### `MAG-BENCH-002A` — pure benchmark/cutoff/visibility contracts

No network/source ingestion. Canonical identities and adversarial fixtures only.

### `MAG-BENCH-002B` — contamination/leakage graph

Compose MAT-009/MAT-010 search memory and source/model lineage into target eligibility decisions without one scalar contamination flag.

### `MAG-BENCH-002C` — Fe-Co-X family manifests

Add separately captured family/source manifests with exact source authority. Keep expected answers sealed from candidate generation.

### `MAG-BENCH-003` — executable cross-family/temporal/structural holdout campaign

Only after the contracts qualify, run retrospective benchmarks and preserve complete selection/result census.

## Dependencies

- MAG-001 #4347
- MAG-002 #4437
- MAT-007 #4336
- MAT-008 #4339
- MAT-009 #4340
- MAT-010 #4341
- MAT-011 / MAT-011B
- MAT-016 firewall/admission
- MAT-CALC-001 #5114 for local high-fidelity execution evidence
- THERMO-001 #5015
- PHONON-001 #5016

## Claim ceiling

A qualified MAG-BENCH-002 establishes the identity, cutoff, leakage classification and split semantics of a rare-earth-free magnet benchmark. It does not establish model superiority, prospective discovery, material novelty, synthesis, or permanent-magnet usefulness.
